app/components/entity_mentions/models.py

```diff
@@ -0,0 +1,43 @@
+import datetime as dt
+from typing import Annotated, Literal
+
+from pydantic import BaseModel, BeforeValidator, Field
+
+
+def state_validator(v: object) -> bool:
+    if isinstance(v, bool):
+        return v
+    if not isinstance(v, str):
+        msg = "`closed` must be a string or a bool"
+        raise TypeError(msg)
+    if v not in {"open", "closed"}:
+        msg = "`closed` must be 'open' or 'closed'"
+        raise ValueError(msg)
```

Comment:
Q: Couldnt this be a switch/match statement?

Reply:
It could! And it would look nicer as well, will refactor.